# HG changeset patch
# User holger krekel <holger@merlinux.eu>
# Date 1412585605 -7200
# Node ID c9303c302995a4fb844553d4977c55abc72e2020
# Parent  47ff3652b0aed191f7bafc741c211d1415f65224
fix issue35: adapt some doctests, fix some channel tests for py3.
Also modernize tox.ini a bit.
--rebased
diff -ur execnet-1.2.0.orig/doc/example/test_group.txt execnet-1.2.0/doc/example/test_group.txt
--- doc/example/test_group.txt	2014-01-29 19:50:22.000000000 +0800
+++ doc/example/test_group.txt	2014-10-10 19:51:00.108828123 +0800
@@ -14,7 +14,7 @@
     >>> group
     <Group ['gw0', 'gw1']>
     >>> list(group)
-    [<PopenGateway id='gw0' receive-live, 0 active channels>, <PopenGateway id='gw1' receive-live, 0 active channels>]
+    [<Gateway id='gw0' receive-live, thread model, 0 active channels>, <Gateway id='gw1' receive-live, thread model, 0 active channels>]
     >>> 'gw0' in group and 'gw1' in group
     True
     >>> group['gw0'] == group[0]
@@ -37,7 +37,7 @@
     >>> gw = group.makegateway("popen//id=sub1")
     >>> assert gw.id == "sub1"
     >>> group['sub1']
-    <PopenGateway id='sub1' receive-live, 0 active channels>
+    <Gateway id='sub1' receive-live, thread model, 0 active channels>
 
 Getting (auto) IDs before instantiation
 ------------------------------------------------------
diff -ur execnet-1.2.0.orig/doc/example/test_proxy.txt execnet-1.2.0/doc/example/test_proxy.txt
--- doc/example/test_proxy.txt	2014-01-29 19:50:22.000000000 +0800
+++ doc/example/test_proxy.txt	2014-10-10 19:16:50.380156509 +0800
@@ -18,10 +18,10 @@
     >>> group.defaultspec = 'popen//via=master'
     >>> master = group.makegateway('popen//id=master')
     >>> master
-    <Gateway id='master' receive-live, 0 active channels>
+    <Gateway id='master' receive-live, thread model, 0 active channels>
     >>> slave = group.makegateway()
     >>> slave
-    <Gateway id='gw0' receive-live, 0 active channels>
+    <Gateway id='gw0' receive-live, thread model, 0 active channels>
     >>> group
     <Group ['master', 'gw0']>
 
diff -ur execnet-1.2.0.orig/testing/test_channel.py execnet-1.2.0/testing/test_channel.py
--- testing/test_channel.py	2014-01-29 19:50:22.000000000 +0800
+++ testing/test_channel.py	2014-10-10 19:21:00.943955387 +0800
@@ -68,25 +68,19 @@
 
     def test_channel_error_reporting(self, gw):
         channel = gw.remote_exec('def foo():\n  return foobar()\nfoo()\n')
-        try:
-            channel.receive()
-        except channel.RemoteError:
-            e = sys.exc_info()[1]
-            assert str(e).startswith('Traceback (most recent call last):')
-            assert str(e).find('NameError: global name \'foobar\' '
-                               'is not defined') > -1
-        else:
-            pytest.fail('No exception raised')
+        excinfo = pytest.raises(channel.RemoteError, channel.receive)
+        msg = str(excinfo.value)
+        assert msg.startswith('Traceback (most recent call last):')
+        assert "NameError" in msg
+        assert "foobar" in msg
 
     def test_channel_syntax_error(self, gw):
         # missing colon
         channel = gw.remote_exec('def foo()\n return 1\nfoo()\n')
-        try:
-            channel.receive()
-        except channel.RemoteError:
-            e = sys.exc_info()[1]
-            assert str(e).startswith('Traceback (most recent call last):')
-            assert str(e).find('SyntaxError') > -1
+        excinfo = pytest.raises(channel.RemoteError, channel.receive)
+        msg = str(excinfo.value)
+        assert msg.startswith('Traceback (most recent call last):')
+        assert "SyntaxError" in msg
 
     def test_channel_iter(self, gw):
         channel = gw.remote_exec("""
diff -ur execnet-1.2.0.orig/tox.ini execnet-1.2.0/tox.ini
--- tox.ini	2014-01-29 19:50:22.000000000 +0800
+++ tox.ini	2014-10-10 20:14:22.806301771 +0800
@@ -1,16 +1,11 @@
 [tox]
-envlist=py26,py27,py32,py33,pypy,flakes
-distshare={homedir}/.tox/distshare
-
-[tox:hudson]
-distshare={toxworkdir}/distshare
-sdistsrc={distshare}/execnet-*
+envlist=py26,py27,py33,py34,pypy,flakes
 
 [testenv]
 changedir=testing
 deps=pytest
      pytest-timeout
-commands=py.test -rsfxX --junitxml={envlogdir}/junit-{envname}.xml []
+commands=py.test -rsfxX {posargs}
 
 [testenv:flakes]
 changedir=
@@ -26,8 +21,7 @@
 deps=sphinx
      pytest
 commands=
-    py.test \
-        -rsfxX -v --junitxml={envlogdir}/junit-{envname}.xml check_sphinx.py
+    py.test  -rsfxX -v check_sphinx.py
 
 [pytest]
 timeout = 20
